WcaOnRails/app/models/competition.rb

```diff
@@ -427,16 +439,24 @@ def dangerously_close_to?(c)
     self.kilometers_to(c) <= NEARBY_DISTANCE_KM_DANGER && days_until.abs < NEARBY_DAYS_DANGER
   end
 
+  def results_uploaded?
+    Result.exists?(competitionId: self.id)
+  end
+
   def user_can_view?(user)
     self.showAtAll || (user && user.can_manage_competition?(self))
   end
 
+  def in_progress?
+    start_date <= Date.today && end_date >= Date.today
```

Comment:
`.cover?` looks nice. Good catch about the duplicate calls to `Date.today` =)

Can you explain the second one? I don't get it.

```ruby
~ @kaladin> irb
irb(main):001:0> (1..5) == 4
=> false
irb(main):002:0> (1..5) == 6
=> false
irb(main):003:0> 
```